third_party/libadblockplus/grit/resources/elemhide_for_selector.jst

```diff
@@ -0,0 +1,9 @@
+// {{filename_with_query}} and {{url}} will be replaced with actual variable values
+let selector = "[src$='{{filename_with_query}}'], [srcset$='{{filename_with_query}}']";
+let elements = document.querySelectorAll(selector);
+if (elements.length > 0) {
+  for (let element of elements) {
+   if (element.src == '{{url}}')
```

Comment:
Is somewhere here a possibility for JS injection?
Consider, e.g. url like `https://some/url/image' && alert("something-evil") && '' == '`.
E.g. we could take a look at https://github.com/joliss/js-string-escape/blob/master/index.js and sanitize the string in C++ before passing it to `replaceString`.
cc=@hfiguiere, @shoniko do you have some recommendations?

Reply:
i've updated the code and now escaping `filename_with_query`. there is no need to escape `url` as it comes from chromium and it's canonical.

Reply:
i'd better use the world "valid":
https://cs.chromium.org/chromium/src/url/gurl.h?q=url.h&sq=package:chromium&g=0&l=112

so here i've  meant there is no JS injected here.

Reply:
sure, https://gitlab.com/eyeo/adblockplus/chromium/issues/12#note_97530896